contracts/Core/AssetManager.sol

```diff
@@ -306,6 +282,124 @@ contract AssetManager is ACL, AssetStorage {
+                numAssets = numAssets + 1;
+
+                collections[numAssets].id = numAssets; 
+                collections[numAssets].name = pendingCollections[i].name; 
+                collections[numAssets].aggregationMethod = pendingCollections[i].aggregationMethod; 
+                collections[numAssets].epoch = currentEpoch;
+                collections[numAssets].creator = pendingCollections[i].creator;
+                collections[numAssets].repeat = pendingCollections[i].repeat;
+                collections[numAssets].assetType = uint256(assetTypes.Collection);
+                collections[numAssets].active = true;
+                uint256[] memory jobIDs = pendingCollections[i].jobIDs;
+                for(uint256 j = 0; j < jobIDs.length; j++){
+                    collections[numAssets].jobIDs.push(jobIDs[j]);
+                    collections[numAssets].jobIDExist[jobIDs[j]] = true;
+                    pendingCollections[i].jobIDExist[jobIDs[j]] = false;
+                }
+                
+                delete (pendingCollections[i]);
+
+                numPendingCollections = numPendingCollections - 1;
+
+                activeAssets.push(numAssets);
+                
+                emit CollectionActivityStatus(
+                    numAssets,
+                    currentEpoch,
+                    collections[numAssets].active,
+                    block.timestamp
+                );
+
+            }    
+        }
+    }
+
+    function deactivateAssets() 
+        external
+        onlyRole(parameters.getAssetConfirmerHash()) 
+    {
+        uint256 epoch = parameters.getEpoch();
+        
+        for(uint256 i = 0; i < pendingAssetDeactivation.length; i++){
+            uint256 id = pendingAssetDeactivation[i];
+
+            if(jobs[id].assetType == uint256(assetTypes.Job)){
+                jobs[id].active = false;
+                emit JobActivityStatus(
+                id,
+                epoch,
+                jobs[id].active,
+                block.timestamp
+                );
+            }
+            else{
+                collections[id].active = false;
+
+                for(uint j = 0; j < activeAssets.length; j++){
+                    if(id == activeAssets[j]){
+                        activeAssets[j] = activeAssets[activeAssets.length - 1];
+                        activeAssets.pop();
+                        break;
+                    }
+                }
+
+                emit CollectionActivityStatus(
+                id,
+                epoch,
+                collections[id].active,
+                block.timestamp
+                );
+            }
+        }
+        delete(pendingAssetDeactivation);
+    }
+
+    function activateAssets() 
+        external
+        onlyRole(parameters.getAssetConfirmerHash()) 
+    {
+        uint256 epoch = parameters.getEpoch();
+        
+        for(uint256 i = 0; i < pendingAssetActivation.length; i++){
```

Comment:
what if admin creates a large number of jobs by mistake? (50+)

Reply:
They can be deactivated then. Anyways, stakers wont be reporting for jobs anymore, only collections will reported on so it does not affect reporting also

Reply:
what if its not deactivated?
to be on safer side, there should be a cap on how many jobs/collections added in single epoch. and throw if additional jobs are added